src/vcf/validator.cpp

```diff
@@ -20,6 +20,12 @@ namespace ebi
 {
   namespace vcf
   {
+    bool validate(const std::vector<char> &firstLine,
```

Comment:
I put this two in the cpp to hide them from public access (we agreed that only is_valid_vcf_file should be provided). As build_parser and detect_version are not used from the tests, I could move those 2 to the cpp as well to be a bit more consistent. Full public hpp or private declaration?